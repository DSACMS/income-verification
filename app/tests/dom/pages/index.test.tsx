// test/pages/index.test.js
import { act, render, screen } from "@testing-library/react";
import { axe } from "jest-axe";

import Index from "../../../src/pages/index";

describe("Index", () => {
  it("should render the heading", () => {
    render(<Index />);

    const heading = screen.getByText(/Upload documents/i);

    expect(heading).toBeInTheDocument();
    expect(heading).toMatchSnapshot();
  });

  it("should pass accessibility scan", async () => {
    const { container } = render(<Index />);
    await act(async () => {
      const results = await axe(container);

      expect(results).toHaveNoViolations();
    });
  });
});
