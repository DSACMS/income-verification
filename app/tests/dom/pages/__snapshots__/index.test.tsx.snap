// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Index > should render the heading 1`] = `
<button
  class="usa-button"
  data-testid="button"
  type="button"
>
  Upload documents
</button>
`;

exports[`Index should render the heading 1`] = `
<button
  class="usa-button"
  data-testid="button"
  type="button"
>
  Upload documents
</button>
`;
